File: Java/Sorting/src/sort/methods/InsertionSort.java
package sort.methods;

import java.util.ArrayList;

public class InsertionSort {

    public ArrayList<Integer> InsertionSort(ArrayList arr) {
        /** Iterate over each element **/
        for (int i = 0; i < arr.size(); i++) {
            /** For each element seen we want to check if values on it's left are less than itself
             *  Key will be the element in question
             *  j will be the elements left of key using a while loop
             * **/
            Integer key = (Integer) arr.get(i);
            int j = i - 1;

            /** So if the key is greater than left element we need to shift elements to the right opening a gap **/
            while ( j >= 0 && key < (Integer) arr.get(j)) {
                arr.set(j + 1, arr.get(j));
                j--;
            }
            arr.set(j + 1, key);
        }
        return arr;
    }
}

File: JS/InsertionSort.js

let insertionSort = (arr) => {
    // Each element iterated over is a "key"
    // Take the key and compare to values on it's left
    //      If the key is less than it's left we swap them until they are not
    for (let i = 1; i < arr.length; i++) {
        let key = arr[i];
        let j = i - 1;

        while (j > 0 && key < arr[j]) {
            arr[j + 1] = arr[j];
            j = j - 1;
        }
        arr[j + 1] = key;
    }
}